.github/workflows/go.yml

```diff
@@ -29,7 +29,7 @@ jobs:
 
   test-unit:
     needs: check-changes
-    if: ${{ needs.check-changes.outputs.has_changes == 'yes' }}
+    if: ${{ needs.check-changes.outputs.has_changes == 'yes' && matrix.os != '' }}
```

Comment:
the build is failed, https://github.com/antrea-io/antrea/actions/runs/2068769348
I am confused about the purpose of this change considering `matrix.os` is always configured in this job.

Reply:
Thanks for review. I was trying a workaround to fix https://github.com/antrea-io/antrea/issues/3563. It turned out to not working. I will close the PR and discuss solution there first.